api/v1alpha1/clienttrafficpolicy_types.go

```diff
@@ -132,10 +132,18 @@ type HeaderSettings struct {
 	// PreserveXRequestID configures Envoy to keep the X-Request-ID header if passed for a request that is edge
 	// (Edge request is the request from external clients to front Envoy) and not reset it, which is the current Envoy behaviour.
 	// It defaults to false.
+	// Deprecated: use RequestID instead
```

Comment:
need to mention what happens when both are set. Or, we should validate only either of them is set at CEL validation.

Reply:
also it'd be more user friendly how they can use RequestID to configure the same. (i.e. set `Preserve`)